compiler/rustc_trait_selection/src/solve/search_graph/mod.rs

```diff
@@ -47,7 +50,7 @@ pub(super) struct SearchGraph<'tcx> {
     ///
     /// An element is *deeper* in the stack if its index is *lower*.
     stack: IndexVec<StackDepth, StackEntry<'tcx>>,
-    provisional_cache: ProvisionalCache<'tcx>,
+    stack_entries: FxHashMap<CanonicalInput<'tcx>, StackDepth>,
```

Comment:
side-note: maybe we should use an FxIndexMap to prevent that query instability lint.